### ISSUE TITLE
Htmlmixed mode does not correctly color custom elements

### ISSUE THREAD

**Issue by TimvdLippe** [codemirror/CodeMirror#4563]
Custom elements is a new spec that has been accepted and is currently being implemented by browsers. Custom elements require a `-` inside the name, for example `<paper-button></paper-button>`. Currently htmlmixed mode does not detect these tags. See https://github.com/webcomponents/webcomponents.org/issues/859 for usage of CodeMirror breaking highlighting of these tags. Instead of all red, the text before the `-` is red and afterwards orange. The expected behavior is that the whole name is red.

**Comment by abdonrd** [comment]
Thanks for the fix, @marijnh! When a new release? 🙄

**Comment by adrianheine** [comment]
@marijnh usually cuts one release per month, as you can see from the [tag overview](https://github.com/codemirror/CodeMirror/tags).

**Comment by abdonrd** [comment]
Thanks @adrianheine!

**Comment by samuelli** [comment]
Requesting that 81b7760 be reverted. The regex is an inverted character set match. Code mirror was actually functioning correctly. The fix actually breaks it instead.

**Comment by marijnh** [comment]
Argh. Reverted as 8a52fcee

@TimvdLippe What exactly were you testing before?

**Comment by TimvdLippe** [comment]
@marijnh Sorry for the confusion. I am a user of webcomponents.org and I was following the comment of @samuelli : https://github.com/webcomponents/webcomponents.org/issues/859#issuecomment-278497307 Therefore I encountered this issue on the website and thought that CodeMirror was the culprit. However after updating, @samuelli discovered it was not CodeMirror that was the problem, but the GitHub markdown highlighter which is ALSO used in the project. The usage of the two syntax highlighters confused us. I am very sorry for the trouble we caused you!
